pandas/core/frame.py

```diff
@@ -3207,13 +3206,8 @@ def _set_value(self, index, col, value, takeable: bool = False):
                 return
 
             series = self._get_item_cache(col)
-            engine = self.index._engine
-            loc = engine.get_loc(index)
-            validate_numeric_casting(series.dtype, value)
-
-            series._values[loc] = value
-            # Note: trying to use series._set_value breaks tests in
-            #  tests.frame.indexing.test_indexing and tests.indexing.test_partial
+            self.index._engine.get_loc(index)
```

Comment:
why are you discarding `loc`?

Reply:
If get_loc raises a ``KeyError`` we have to catch this here insted of in ``series._set_value`` but we do not need ``loc``

Reply:
why bother calling get_loc here at all if its going to be called again in series._set_value?

Reply:
We want to arrive in the except clause here, when index does not exist instead of the except clause in `` series._set_value``

Reply:
> We want to arrive in the except clause here, when index does not exist instead of the except clause in series._set_value

can you elaborate on this?  is it bc the except clause here does a takeable check whereas the one in _set_value does not?  if so, could just add that check there?

why is the deleted comment about "tests.frame.indexing.test_indexing and tests.indexing.test_partial" no longer relevant?